### PR TITLE
Update helm chart to accomodate recent project changes

### PR DESCRIPTION
**Pull request by ShaunMaher**
* Remove the `--run_gunicorn ` startup flag
* Tidy up some helm chart build edge cases (i.e. can't increment version if there is no previous version)
* Update build process to use the new chart name (`-helm` suffix removed).  Thanks to @Lunik for forcing me to reevaluate my previous assumption about the helm chart name conflicting with the image name.  It seems that it "just works".
* Integrate helm chart build process into the process that builds the docker images so that this becomes a single step (#2040)
* Tweak the GitHub workflow that builds the docker images so it only tries to push to docker hub if the project running the build is `BerriAI/litellm`.  This allows other developers to fork this repository and test the workflows on their forks, without them always failing because they (obviously) can't push to docker hub as BerriAI.
* This should also close the deployment parts of #1880, but mainly because @Lunik's work on #2018

### CODE REVIEW

**Comment by ishaan-jaff** [comment]
cc @Lunik can we get your feedback on this too - since you use this ?

**Comment by debdutdeb** [comment]
Would be amazing to have this merged  🙏🏼  I'd rather not copy the whole stuff to another chart to use as dependency. 

Please let me know if I can help. 

From the looks of it, structurally it looks all good, but I've not looked deep in.

**Comment by ishaan-jaff** [comment]
@debdutdeb it would help us if you could check if this helm chart works for you ?

File: deploy/charts/litellm/templates/deployment.yaml [hunk 1]
```diff
@@ -161,7 +161,6 @@ spec:
           args:
             - --config
             - /etc/litellm/config.yaml
-            - --run_gunicorn
```
Comment (ishaan-jaff) [inline deploy/charts/litellm/templates/deployment.yaml:164]: why remove this ?

**Review by ishaan-jaff — CHANGES_REQUESTED** [review]
re `run_gunicorn`

**Comment by ishaan-jaff** [comment]
working on getting these changes merged now !

**Comment by ishaan-jaff** [comment]
Update I took these changes and added it to main, confirmed it works. We needed to rename the chart to `litellm-helm` since it would overwrite the litellm docker image 

I'm going to still merge this PR and then re-apply my changes. Want to make sure I recognize @ShaunMaher's work on our contributors list + release notes !

**Pull request merged**